### PR TITLE
Add tests for Aggregator constructors

### PR DESCRIPTION
**Pull request by gmponos**
According to the tests you can see here is this the expected behavior?

### CODE REVIEW

**Comment by frederikbosch** [comment]
@gmponos Obviously not. Passing in an empty array means 100% sure to have a failing parser/formatter object. Therefore I would not find it breaking to throw a `InvalidArgumentException` on such occasion.

**Comment by gmponos** [comment]
OK... i'll fix this

File: src/Formatter/AggregateMoneyFormatter.php [hunk 1]
```diff
@@ -23,6 +23,10 @@
      */
     public function __construct(array $formatters)
     {
+        if (empty($parsers)) {
```
Comment (slavcodev) [inline src/Formatter/AggregateMoneyFormatter.php]: Do you mean `empty($formatters)`?

Comment (gmponos) [reply src/Formatter/AggregateMoneyFormatter.php]: pfff.. copy-paste!

File: src/Formatter/AggregateMoneyFormatter.php [hunk 2]
```diff
@@ -23,6 +23,10 @@
      */
     public function __construct(array $formatters)
     {
+        if (empty($formatters)) {
+            throw new \InvalidArgumentException(sprintf('Initialize an empty %s is not possible', self::class));
```
Comment (frederikbosch) [inline src/Formatter/AggregateMoneyFormatter.php]: Can you change `self::class` in `get_class($this)`? We are supporting PHP 5.5. Which I do not like, but until we are dropping it, we have to take care of it.

File: src/Parser/AggregateMoneyParser.php [hunk 3]
```diff
@@ -22,6 +22,10 @@
      */
     public function __construct(array $parsers)
     {
+        if (empty($parsers)) {
+            throw new \InvalidArgumentException(sprintf('Initialize an empty %s is not possible', self::class));
```
Comment (frederikbosch) [inline src/Parser/AggregateMoneyParser.php]: Can you change `self::class` in `get_class($this)`? We are supporting PHP 5.5. Which I do not like, but until we are dropping it, we have to take care of it.

File: src/Formatter/AggregateMoneyFormatter.php (hunk 2 above)
Comment (gmponos) [reply src/Formatter/AggregateMoneyFormatter.php]: I am not sure I understand.

http://php.net/manual/en/language.oop5.constants.php

> The special ::class constant are available as of PHP 5.5.0, and allows for fully qualified class name resolution at compile, this is useful for namespaced classes: 

Also the tests are passing for PHP 5.5 also

Comment (frederikbosch) [reply src/Formatter/AggregateMoneyFormatter.php]: Oh right. Damn. My bad.

**Pull request merged**